packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -1733,6 +1790,34 @@ class RenderPhysicalModel extends _RenderPhysicalModelBase<RRect> {
   }
 }
 
+PhysicalModelLayer _createOrUpdatePhysicalModelLayer({
+  @required Path clipPath,
+  @required Clip clipBehavior,
+  @required double elevation,
+  @required Color color,
+  @required Color shadowColor,
+  @required Rect childPaintBounds,
+  @required PhysicalModelLayer oldLayer,
+}) {
+  if (oldLayer == null) {
+    return PhysicalModelLayer(
+      clipPath: clipPath,
+      clipBehavior: clipBehavior,
+      elevation: elevation,
+      color: color,
+      shadowColor: shadowColor,
+    );
+  } else {
+    oldLayer
+      ..clipPath = clipPath
+      ..clipBehavior = clipBehavior
+      ..elevation = elevation
+      ..color = color
+      ..shadowColor = shadowColor;
+    return oldLayer;
+  }
+}
```

Comment:
here and in other places, might it be better to just create the layer using `oldLayer ??= PhysicalModelLayer()` and then set the attributes unconditionally? Do the constructors do anything more efficient than that?

Reply:
We can give them default values, or we can allow null and document they have to be non-null before adding to scene and add the relevant asserts (we have some cases like that already). I think the ergonomics will be nicer if we remove the `@required`s and just have the one way of doing this, though. Having both ways is going to result in bugs when we accidentally only fix one code path.